src/plots/linePlot.ts

```diff
@@ -349,10 +374,65 @@ export module Plots {
 
     protected _getDataToDraw() {
       let dataToDraw = new Utils.Map<Dataset, any[]> ();
-      this.datasets().forEach((dataset) => dataToDraw.set(dataset, [dataset.data()]));
+
+      this.datasets().forEach((dataset) => {
+        let data = dataset.data();
+
+        if (!this._croppedRendering) {
+          dataToDraw.set(dataset, [data]);
+          return;
+        }
+
+        let filteredDataIndices = data.map((d, i) => i);
+
+        if (this._croppedRendering) {
+          filteredDataIndices = this._filterCroppedRendering(dataset, filteredDataIndices);
+        }
+
+        dataToDraw.set(dataset, [filteredDataIndices.map((d, i) => data[d])]);
```

Comment:
Curious why you didn't use a `filter` code flow instead?

Have the indices exist in a set and then filter if the index exists in the set?